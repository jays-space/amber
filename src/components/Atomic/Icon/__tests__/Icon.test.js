import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Icon } from "..";

describe("<Icon />", () => {
  it("should render correctly", () => {
    render(<Icon name="add" />);

    const iconElement = screen.getByTestId("icon-add");
    expect(iconElement).toMatchSnapshot();
  });

  it("should pass custom classes into the element's classes", () => {
    const customClass = "custom-class";
    render(<Icon name="add" className={customClass} />);

    const iconElement = screen.getByTestId("icon-add");
    expect(iconElement.classList).toContain(customClass);
  });

  // test all branches
  describe("it renders the correct icon", () => {
    const iconNames = [
      "filter",
      "download",
      "print",
      "add",
      "more",
      "home",
      "inbox",
      "calendar",
      "task",
      "project",
      "new",
      "person",
      "people",
      "report",
      "settings",
      "minimize",
      "message",
      "email",
      "previous",
      "next",
      "back",
      "notification",
      "visible",
      "hidden",
      "search",
      "menu",
      "spinner",
    ];

    for (let icon of iconNames) {
      it("should render the add icon component", () => {
        render(<Icon name={icon} />);

        const iconElement = screen.getByTestId(`icon-${icon}`);
        expect(iconElement).toBeInTheDocument();
      });
    }
  });
});
