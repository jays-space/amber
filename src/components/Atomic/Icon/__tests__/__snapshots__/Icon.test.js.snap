// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> should render correctly 1`] = `
<svg
  class="text-slate-500 "
  data-testid="icon-add"
  fill="currentColor"
  height="24"
  stroke="currentColor"
  stroke-width="0"
  viewBox="0 0 512 512"
  width="24"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M416 277.333H277.333V416h-42.666V277.333H96v-42.666h138.667V96h42.666v138.667H416v42.666z"
  />
</svg>
`;
